frontend/src/metabase/common/components/DataPicker/components/DataPickerModal.tsx

```diff
@@ -0,0 +1,61 @@
+import { useCallback } from "react";
+import { t } from "ttag";
+import _ from "underscore";
+
+import type { CollectionId, TableId } from "metabase-types/api";
+
+import type { EntityPickerModalOptions, EntityTab } from "../../EntityPicker";
+import { EntityPickerModal, defaultOptions } from "../../EntityPicker";
+import type { NotebookDataPickerValueItem, TablePickerValue } from "../types";
+
+import { TablePicker } from "./TablePicker";
+
+interface Props {
+  /**
+   * TODO: use this prop in https://github.com/metabase/metabase/issues/40719
+   */
+  collectionId: CollectionId | null | undefined;
+  value: TablePickerValue | null;
+  onChange: (value: TableId) => void;
+  onClose: () => void;
+}
+
+const options: EntityPickerModalOptions = {
+  ...defaultOptions,
+  hasConfirmButtons: false,
+};
+
+export const DataPickerModal = ({ value, onChange, onClose }: Props) => {
+  const handleItemChange = useCallback(
+    (item: NotebookDataPickerValueItem) => {
+      onChange(item.id);
+      onClose();
+    },
+    [onChange, onClose],
+  );
+
+  const tabs: [
+    EntityTab<NotebookDataPickerValueItem["model"]>,
+    ...EntityTab<NotebookDataPickerValueItem["model"]>[],
+  ] = [
+    {
+      displayName: t`Tables`,
+      model: "table",
+      icon: "table",
+      element: <TablePicker value={value} onChange={handleItemChange} />,
+    },
+  ];
+
+  return (
+    <EntityPickerModal
+      canSelectItem
+      options={options}
+      selectedItem={null}
+      tabs={tabs}
+      title={t`Pick your starting data`}
+      onClose={onClose}
+      onConfirm={_.noop} // onConfirm is unused when options.hasConfirmButtons is falsy
```

Comment:
so can't we pass undefined instead and handle it on EntrityPickerModal level?

Reply:
We could. I think it's probably better to get rid of `options.hasConfirmButtons` and make it depend on the presence of `onConfirm` prop though. It might be a controversial change so I'll submit it as a separate PR: #41172